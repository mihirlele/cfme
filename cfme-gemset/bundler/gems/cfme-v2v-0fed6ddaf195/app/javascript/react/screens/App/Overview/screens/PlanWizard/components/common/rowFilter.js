/**
 * Simple client side row filter for VM table
 *
 * activeFilters: {
 *  label: 'VM Name: {value}',
 *  field: {id: "name", title: "VM Name", placeholder: "Filter by VM Name", filterType: "text"},
 *  value: '{value}'
 * }
 *
 * rows: [{
 *  name: 'cfmetest67',
 *  path: 'vCenter/Datacenter',
 *  cluster: 'Raleigh'
 *  ...
 * }]
 */
export default function rowFilter(activeFilters, rows) {
  if (activeFilters && activeFilters.length && rows && rows.length) {
    const filteredRows = [];
    rows.forEach(row => {
      const match = activeFilters.every(
        filter =>
          row[filter.field.id] && filter.value
            ? `${row[filter.field.id].toLowerCase()}`.indexOf(filter.value.toLowerCase()) > -1
            : null
      );
      if (match) {
        filteredRows.push(row);
      }
    });
    return filteredRows;
  }
  return rows;
}
