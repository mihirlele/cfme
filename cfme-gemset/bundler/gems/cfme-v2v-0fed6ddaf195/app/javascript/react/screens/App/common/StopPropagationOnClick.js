import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

// This component has no role on the page, and will be hidden from screen readers (role="presentation").
// Prevents a click on an inner element (e.g. a DropdownKebab) from triggering a click on an ancestor (e.g. a table row).
// The onKeyUp handler and role are used to satisfy eslint-plugin-jsx-a11y.
const StopPropagationOnClick = ({ children }) => (
  <span onClick={e => e.stopPropagation()} onKeyUp={noop} role="presentation">
    {children}
  </span>
);

StopPropagationOnClick.propTypes = {
  children: PropTypes.node
};

export default StopPropagationOnClick;
