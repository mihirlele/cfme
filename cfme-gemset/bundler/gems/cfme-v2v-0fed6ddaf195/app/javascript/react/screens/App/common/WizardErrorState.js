import React from 'react';
import PropTypes from 'prop-types';

const WizardErrorState = ({ title, message, onClose }) => (
  <div className="wizard-pf-complete blank-slate-pf">
    <div className="wizard-pf-success-icon">
      <span className="pficon pficon-error-circle-o" />
    </div>
    <h3 className="blank-slate-pf-main-action">{title}</h3>
    <p className="blank-slate-pf-secondary-action">{message}</p>
    {onClose ? (
      <button type="button" className="btn btn-lg btn-primary" onClick={onClose}>
        {__('Close')}
      </button>
    ) : null}
  </div>
);

WizardErrorState.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string,
  onClose: PropTypes.func
};

WizardErrorState.defaultProps = {
  message: ''
};

export default WizardErrorState;
