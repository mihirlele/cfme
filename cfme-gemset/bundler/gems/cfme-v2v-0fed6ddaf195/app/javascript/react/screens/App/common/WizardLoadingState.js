import React from 'react';
import PropTypes from 'prop-types';
import { Spinner } from 'patternfly-react';

const WizardLoadingState = ({ title, message }) => (
  <div className="wizard-pf-process blank-slate-pf">
    <Spinner loading size="lg" className="blank-slate-pf-icon" />
    <h3 className="blank-slate-pf-main-action">{title}</h3>
    <p className="blank-slate-pf-secondary-action">{message}</p>
  </div>
);

WizardLoadingState.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string
};

WizardLoadingState.defaultProps = {
  message: ''
};

export default WizardLoadingState;
