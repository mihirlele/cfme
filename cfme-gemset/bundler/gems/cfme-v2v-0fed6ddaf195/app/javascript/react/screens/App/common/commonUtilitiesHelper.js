const commonUtilitiesHelper = {
  getMostRecentEntityByCreationDate: entities =>
    entities && entities.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current)),

  groupBy: (items, key) =>
    items &&
    items.reduce(
      (result, item) => ({
        ...result,
        [item[key]]: [...(result[item[key]] || []), item]
      }),
      {}
    ),

  flatten: (arrayItems, flattenedArray) =>
    arrayItems && arrayItems.map(task => task.map(arrayElement => flattenedArray.push(arrayElement)))
};

export default commonUtilitiesHelper;
