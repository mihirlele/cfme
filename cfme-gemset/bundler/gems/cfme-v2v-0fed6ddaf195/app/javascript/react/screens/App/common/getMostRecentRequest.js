const getMostRecentRequest = requests =>
  requests.length && requests.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current));

export default getMostRecentRequest;
