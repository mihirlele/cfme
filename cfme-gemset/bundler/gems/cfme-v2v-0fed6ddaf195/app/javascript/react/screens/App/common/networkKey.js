/**
 * returns a unique network based on current UI model
 */
export const networkKey = network => `${network.clusterId}\\${network.uid_ems || network.ems_ref}`;
