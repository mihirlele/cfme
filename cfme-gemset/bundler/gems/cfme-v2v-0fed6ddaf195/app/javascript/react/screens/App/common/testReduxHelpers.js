import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import configureMockStore from 'redux-mock-store';

const middlewares = [thunk, promiseMiddleware()];

export const generateStore = (reducers, initialState) =>
  createStore(combineReducers(reducers), initialState, applyMiddleware(...middlewares));

export const mockStore = initialState => configureMockStore(middlewares)(initialState);
