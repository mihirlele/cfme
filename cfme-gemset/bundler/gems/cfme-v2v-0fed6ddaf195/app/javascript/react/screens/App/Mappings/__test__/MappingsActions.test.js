import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';

import { mockRequest, mockReset } from '../../../../../common/mockRequests';
import { fetchCloudTenantsAction } from '../MappingsActions';

const middlewares = [thunk, promiseMiddleware()];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

afterEach(() => {
  store.clearActions();
  mockReset();
});

describe('fetchCloudTenantsAction', () => {
  test('dispatches PENDING and FULFILLED actions', () => {
    mockRequest({
      method: 'GET',
      url: '/api',
      status: 200,
      response: { mock: 'data' }
    });

    return store.dispatch(fetchCloudTenantsAction('/api')).then(() => {
      const actions = store.getActions();
      expect(actions).toMatchSnapshot();
    });
  });

  test('dispatches PENDING and REJECTED actions', () => {
    mockRequest({
      method: 'GET',
      url: '/api',
      status: 404,
      response: { error: 'error' }
    });

    return store.dispatch(fetchCloudTenantsAction('/api')).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
