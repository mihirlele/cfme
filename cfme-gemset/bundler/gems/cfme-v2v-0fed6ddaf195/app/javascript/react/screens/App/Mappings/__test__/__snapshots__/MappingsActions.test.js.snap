// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchCloudTenantsAction dispatches PENDING and FULFILLED actions 1`] = `
Array [
  Object {
    "type": "FETCH_CLOUD_TENANTS_PENDING",
  },
  Object {
    "payload": Object {
      "mock": "data",
    },
    "type": "FETCH_CLOUD_TENANTS_FULFILLED",
  },
]
`;

exports[`fetchCloudTenantsAction dispatches PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "type": "FETCH_CLOUD_TENANTS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "FETCH_CLOUD_TENANTS_REJECTED",
  },
]
`;
